### PR TITLE
Refactoring Travis build for CakePHP >= 3.2.x and PHP >= 5.5

### PR DESCRIPTION
**Pull request by madalinignisca**
Hi.

As CakePHP 3.2 requires minimum PHP 5.5.9, I thought that instead of installing/enabling APC to test with Travis, as that comes for >5.5 with OPCACHE enabled by default, we should test against APCu.

Also I refactored the .travis.yml file to check against the $TRAVIS_PHP_VERSION variable available.

I've switch in phpunit.xml.dist from apcu

Build log can be checked at https://travis-ci.org/16nsk/cakephp/builds/120096390

(#14.4 build fails identical like for the master branch, I would like to figure it out next).

### CODE REVIEW

**Comment by ADmad** [comment]
Why have you commented out hhvm builds?

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 language: php
 
 php:
-  - 5.5
-  - 5.6
-  - 7.0
+  - '5.5'
```
Comment (dereuromark) [inline .travis.yml]: Why are those changes necessary?

**Comment by 16nsk** [comment]
@ADmad I enabled them back, didn't know you can enable extra php versions through the "matrix" section.

@dereuromark To improve the Travis file. I learned recently about it and wanted to contribute.

File: .travis.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ services:
 
 cache:
   directories:
-    - vendor
+    # - vendor
```
Comment (ADmad) [inline .travis.yml]: Why is this commented out?

File: .travis.yml [hunk 3]
```diff
@@ -58,16 +57,23 @@ before_script:
...
-  - sh -c "if [ '$HHVM' != '1' ]; then echo 'extension = apc.so' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini; fi"
-  - sh -c "if [ '$PHP' = '5.6' ]; then pecl install apc; fi"
-  - sh -c "if [ '$HHVM' = '1' ]; then composer require lorenzo/multiple-iterator=~1.0; fi"
+  - if [[ $TRAVIS_PHP_VERSION != 'hhvm' ]] ; then echo 'extension = memcached.so' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini; fi
+  - if [[ $TRAVIS_PHP_VERSION != 'hhvm' ]] ; then echo 'extension = redis.so' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini; fi
+  - if [[ $TRAVIS_PHP_VERSION != 'hhvm' ]] ; then echo 'extension = apcu.so' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini; fi
+  - if [[ $TRAVIS_PHP_VERSION =~ 5.[56] ]] ; then echo yes | pecl install apcu-4.0.10; fi
+  - if [[ $TRAVIS_PHP_VERSION = 7.* ]] ; then echo yes | pecl install apcu; fi
+  - if [[ $TRAVIS_PHP_VERSION = 7.* ]] ; then pecl config-set preferred_state beta; echo yes | pecl install apcu_bc; fi
+  - if [[ $TRAVIS_PHP_VERSION != 'hhvm' ]] ; then composer require lorenzo/multiple-iterator=~1.0; fi
```
Comment (ADmad) [inline .travis.yml]: You have inverted the condition. The package needs to be installed if the php version **is** hhvm.

File: .travis.yml (hunk 1 above)
Comment (dereuromark) [inline .travis.yml]: I have never seen it defined as strings before in any repo.
Can you please clarify why you think this is an improvement?
The PR should be limited to only the absolutly required changes to make it easier to be approved and passed, right now there seems to be quite some noise changes on top that do not seem clear without further explanations IMO.

Comment (16nsk) [inline .travis.yml]: Oh, ok then my bad. just made them as the first example on the Travis documentation.

Sometimes if you see as a beginner different then in the official documentation may be confusing.

File: .travis.yml (hunk 2 above)
Comment (16nsk) [inline .travis.yml]: I've put it back. I forgot it as I was trying to figure out why one of the tests was failing.

**Comment by markstory** [comment]
Thanks :+1:

**Comment by 16nsk** [comment]
Thank you as well :8ball:
